src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -99,6 +99,29 @@ RewriteResponse TheoryBuiltinRewriter::postRewrite(TNode node) {
   }
 }
 
+TypeNode TheoryBuiltinRewriter::getFunctionTypeForArrayType( TypeNode atn, Node bvl ) {
+  std::vector<TypeNode> children;
+  for( unsigned i=0; i<bvl.getNumChildren(); i++ ){
+    Assert( atn.isArray() );
+    Assert( bvl[i].getType()==atn.getArrayIndexType() );
```

Comment:
Can this fail in practice, i.e. the count for the atn is different than bvl.getNumChildren()?

Reply:
Yes, it is up to the caller to ensure that they give an appropriate bvl/atn pair.